felix/bpf/proxy/syncer.go

```diff
@@ -194,9 +200,15 @@ func uniqueIPs(ips []net.IP) []net.IP {
 
 // NewSyncer returns a new Syncer
 func NewSyncer(nodePortIPs []net.IP,
-	svcsmap *cachingmap.CachingMap[nat.FrontendKey, nat.FrontendValue],
-	epsmap *cachingmap.CachingMap[nat.BackendKey, nat.BackendValue],
-	affmap maps.Map, rt Routes) (*Syncer, error) {
+	svcsmap *cachingmap.CachingMap[nat.FrontendKeyInterface, nat.FrontendValue],
+	epsmap *cachingmap.CachingMap[nat.BackendKey, nat.BackendValueInterface],
+	affmap maps.Map, rt Routes,
+	newFEK func(addr net.IP, port uint16, protocol uint8) nat.FrontendKeyInterface,
```

Comment:
If the number of constructors gets out of hand you could pass them around as methods on a dummy type that implements a factory interface defined here.